### DISCUSSION TITLE
Accessing auth.uid() in PLV8?

### DISCUSSION THREAD

**Discussion by prescience-data** [supabase/supabase#2986] [Questions] [answered]
I assume the way to access this is something like:
```js
const uid = plv8.execute('SELECT * FROM auth.uid()');
```

But was wondering if there is some hidden variable that is pre-loaded into V8 with the current user's `uid`, or some similar more "inline" method of retrieving the uid?

**Comment by steve-chavez** [comment, marked as answer]
> or some similar more "inline" method of retrieving the uid?

Yes, with `select current_setting('request.jwt.claim.sub', true)`, `auth.uid` is just a wrapper around that

```sql
create or replace function auth.uid() returns uuid as $$
  select nullif(current_setting('request.jwt.claim.sub', true), '')::uuid;
$$ language sql stable;
```

> But was wondering if there is some hidden variable that is pre-loaded into V8

Not specifically to v8. On each transaction those settings are set - only when you use the supabase client libs(like supabase-js) or the REST API.

**Reply by prescience-data** [reply]
Ok thanks for that!

Perhaps as a feature request, that would be cool to one day have the current user pre-injected into each V8 script! 

Which could be called like something similar to:

```js
const plan = plv8.prepare('SELECT * FROM tbl WHERE user_id = $1', [ 'string' ]);
const rows = plan.execute([ plv8.uid() ]);
```
